sandbox-backend/src/main/java/com/exadel/sandbox/team5/service/impl/ReviewServiceImpl.java

```diff
@@ -48,7 +50,13 @@ public void delete(Long id) {
     }
 
     @Override
-    public List<ReviewDto> getReviewsByDiscount(Long id) {
-        return mapper.mapAll(reviewDAO.findAllByDiscountId(id), ReviewDto.class);
+    public Map<Integer, Integer> getReviewsByDiscount(Long id) {
+        Map<Integer, Integer> countRate = reviewDAO.findAllRateByDiscountId(id).stream()
```

Comment:
We also can do that on database level by query then we can decrease a number of loaded records